worksheet.rb

```diff
@@ -23,12 +39,159 @@
+      {
+        date: '3rd Feb 2016',
+        cost: 5,
+        rider_id: 'RD0022',
+        rating: 5
+      },
+      {
+        date: "4th Feb 2016",
+        cost: 10,
+        rider_id: "RD0022",
+        rating: 4
+      },
+      {
+        date: "5th Feb 2016",
+        cost: 20,
+        rider_id: "RD0073",
+        rating: 5
+      }
+    ]
+  },
+  {
+    id: 'DR0001',
+    trips: [
+      {
+        date: "3rd Feb 2016",
+        cost: 10,
+        rider_id: "RD0003",
+        rating: 3
+      },
+      {
+        date: "3rd Feb 2016",
+        cost: 30,
+        rider_id: "RD0015",
+        rating: 4
+      },
+      {
+        date: "5th Feb 2016",
+        cost: 45,
+        rider_id: "RD0003",
+        rating: 2
+      }
+    ]
+  },
+  {
+    id: 'DR0002',
+    trips: [
+      {
+        date: "3rd Feb 2016",
+        cost: 25,
+        rider_id: "RD0073",
+        rating: 5
+      },
+      {
+        date: "4th Feb 2016",
+        cost: 15,
+        rider_id: "RD0013",
+        rating: 1
+      },
+      {
+        date: "5th Feb 2016",
+        cost: 45,
+        rider_id: "RD0003",
+        rating: 2
+      }
+    ]
+  },
+  {
+    id: 'DR0003',
+    trips: [
+      {
+        date: "4th Feb 2016",
+        cost: 5,
+        rider_id: "RD0066",
+        rating: 5
+      },
+      {
+        date: "5th Feb 2016",
+        cost: 50,
+        rider_id: "RD0003",
+        rating: 2
+      },
+      {
+        date: "5th Feb 2016",
+        cost: 35,
+        rider_id: "RD0066",
+        rating: 6
+      }
+    ]
+  }
+]
+
 ########################################################
+def total_rides(driver)
+  return driver[:trips].length
+end
+
+def driver_income(driver)
+  return driver[:trips].sum{ |trip| trip[:cost] }
+end
+
+def driver_average_rating(driver)
+  driver_rating_sum = driver[:trips].sum{ |trip| trip[:rating] }
+  return driver_rating_sum.to_f / driver[:trips].length
+end
 # Step 4: Total Driver's Earnings and Number of Rides
 
+
 # Use an iteration blocks to print the following answers:
+
 # - the number of rides each driver has given
+puts "Total number of rides each driver has given:"
+
+drivers.each do |driver|
+  puts "#{driver[:id]} has given #{total_rides(driver)} rides."
+end
+
 # - the total amount of money each driver has made
-# - the average rating for each driver
+puts
+puts 'Total amount of money each driver has made:'
+
+drivers.each do |driver|
+  ride_cost = driver_income(driver)
+  puts "#{driver[:id]} has made $#{ride_cost}."
+end
+
+
+# - the average rating for each driver #
+
+puts
+drivers.each do |driver|
+  driver_rating_average = driver_average_rating(driver)
+  puts "#{driver[:id]}'s average trip rating is #{driver_rating_average.round(1)}."
+end
+
 # - Which driver made the most money?
-# - Which driver has the highest average rating?
\ No newline at end of file
+puts
+richest_driver = drivers.max_by do |driver|
+  driver_income(driver)
+end
+income_sum = driver_income(richest_driver)
```

Comment:
Great use of an enumerable method. Consider wrapping this functionality into a method.